### PR TITLE
Reopen of HSEARCH-4130 Use docker for Elasticsearch IT

### PR DESCRIPTION
**Pull request by fax4ever**
https://hibernate.atlassian.net/browse/HSEARCH-4130

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -910,7 +910,19 @@ void runBuildOnNode(Closure body) {
 
 void runBuildOnNode(String label, Closure body) {
 	node( label ) {
-		timeout( [time: 1, unit: 'HOURS'], body )
+		sh 'docker container prune -f'
+        sh 'docker image prune -f'
+        sh 'docker network prune -f'
+        sh 'docker volume prune -f'
```
Comment (yrodiere) [inline Jenkinsfile:916]: Please encapsulate this in a script, like it's done on Quarkus (where you probably took this from)? Especially if you need to run it twice...

**Review by yrodiere — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (fax4ever) [reply Jenkinsfile:916]: I would avoid to create a file for this. I could extract a method. What do you think about it?

Comment (yrodiere) [reply Jenkinsfile:916]: I guess it could work, yes.

Comment (yrodiere) [reply Jenkinsfile:916]: The build fails. The first failure is caused by Coveralls (it's down today, apparently), but the other one is caused by a docker prune command returning a non-zero status; I assume that happens when there is nothing to prune.

I'm starting to believe that handling this kind of edge cases would be easier in... say... a shell script? :)

**Comment by yrodiere** [comment]
Merged, thanks

**Pull request merged**